src/parsers/aws.ts

```diff
@@ -122,36 +129,64 @@ export const eventBridgeParser = (requestData, responseData) => {
 };
 
 export const sqsParser = (requestData, responseData) => {
-  const { body: reqBody } = requestData;
+  const { body: reqBody } = requestData || {};
+  const reqHeaders = requestData.headers ? requestData.headers : {};
   const { body: resBody } = responseData || {};
-  const parsedReqBody = reqBody ? parseQueryParams(reqBody) : undefined;
-  const parsedResBody = resBody ? traverse(resBody) : undefined;
-  const resourceName = parsedReqBody ? parsedReqBody['QueueUrl'] : undefined;
-  const awsServiceData = { resourceName };
-  // @ts-ignore
-  awsServiceData.messageId =
-    safeGet(parsedResBody, ['SendMessageResponse', 'SendMessageResult', 'MessageId'], undefined) ||
-    safeGet(
-      parsedResBody,
-      [
-        'SendMessageBatchResponse',
-        'SendMessageBatchResult',
-        'SendMessageBatchResultEntry',
-        0,
-        'MessageId',
-      ],
-      undefined
-    ) ||
-    safeGet(
-      parsedResBody,
-      [
-        'SendMessageBatchResponse',
-        'SendMessageBatchResult',
-        'SendMessageBatchResultEntry',
-        'MessageId',
-      ],
-      undefined
-    );
+  let awsServiceData = {};
+
+  // Note: Currently json protocol is new and not commonly used, so the default case is XML.
+  // In the future when JSON usage is default you my want to switch so JSON is the default and XML
+  // protocol is specifically checked for
+  if (
```

Comment:
Make sure we know to extract the `resourceName` in JSON also from other SQS methods like [sendMessageBatch](https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/SQS.html#sendMessageBatch-property)

Reply:
I tested for this in the ```sqsParser -> happy flow JSON format```, the test is run on a single message & a batch message & the message id & resource name are tested

Reply:
Done, fixed it